### ISSUE TITLE
Error creating VLAN interface in different VDOM with fortinet.fortios.fortios_system_interface

### ISSUE THREAD

**Issue by dimk21** [fortinet-ansible-dev/ansible-galaxy-fortios-collection#327]
**Description:**

I am using the Ansible module fortinet.fortios.fortios_system_interface to create a VLAN interface. The physical interface (an aggregate interface) belongs to a different VDOM than the one where I am trying to create the VLAN interface.

**Issue:**

When I attempt to create the VLAN interface, I receive the following error:

"cli_error": "current vf=vdom-cust:2\nentry not found in datasource\nAttribute 'interface' value 'LAN' checking fail -3\nCommand fail. Return code -3\nCommand fail. Return code 1\n",

In this context, the value 'LAN' refers to the aggregate interface that belongs to the root VDOM.

**Steps to Reproduce:**

Configure an aggregate interface in the root VDOM.
Attempt to create a VLAN interface on this aggregate interface in a different VDOM.
Expected Result:
The VLAN interface should be created successfully on the specified aggregate interface.

**Actual Result:**
The process fails with the error mentioned above.

**Additional Information:**

 "meta": {
    "http_method": "POST",
    "revision": "92081e59341427950df2c2f1d04f1dae",
    "revision_changed": true,
    "old_revision": "b4871b12bf5367c985525c9cebbc4a8e",
    "cli_error": "current vf=vdom-cust:2\nentry not found in datasource\nAttribute 'interface' value 'LAN' checking fail -3\nCommand fail. Return code -3\nCommand fail. Return code 1\n",
    "error": -3,
    "status": "error",
    "http_status": 500,
    "vdom": "global",
    "path": "system",
    "name": "interface",
    "serial": "XYZ",
    "version": "v7.0.15",
    "build": 632

Playbook Task:

- name: Configure interfaces
    fortinet.fortios.fortios_system_interface:
      vdom:  "global"
      state: "present"
      access_token: "XYZ"
      system_interface:
        alias: "{{ vrf_name }}"
        allowaccess: "ping"
        name: "INT_{{ vlan_id }}"
        description: "{{ vrf_name }}"
        interface: "LAN"
        ip: "{{ tn_ip }} 255.255.255.254"
        role: "lan"
        status: "down"
        type: "vlan"
        vlanid: "{{ vlan_id }}"
        vdom: "vdom-cust"
        bfd: "enable"
        bfd_desired_min_tx: 750
        bfd_required_min_rx: 750
        bfd_detect_mult: 3
        device_identification: "disable"
        vrf: "{{ vrf_id }}"

**Workaround:**
Any advice or suggestions on how to work around this issue would be greatly appreciated.

Thank you for your assistance!

**Comment by alagoutte** [comment]
Hi,

With the GUI is working ?

I think, you need to set the aggregate on global vdom

**Comment by dimk21** [comment]
> Hi,
> 
> With the GUI is working ?
> 
> I think, you need to set the aggregate on global vdom

Hey,

yes on the GUI the following is working:
Go to VDOM: Global > Interfaces > New Interface > Type VLAN, Virtual Domain: vdom-cust, Interface "LAN"

And no, I cant set the aggregate on global vdom as it is only for management.

**Comment by alagoutte** [comment]
do you have look on the GUI the API call using for create the interface ?

you have 2 vdom on your playbook (or indent issue ?)

**Comment by MaxxLiu22** [comment]
Hi @dimk21 ,

Thank you for bringing this issue to our attention. It appears that FOS is unable to find the interface named "LAN." I noticed that you are using an alias, which is intended for human readability. When FOS refers to the interface, we need to use the actual name instead of the alias. Could you kindly check if "LAN" is indeed the name of the interface you wish to use? On my side, I was able to create a VLAN interface under a VDOM by using a different VDOM aggregate interface in global mode. If the issue persists, could you please provide a GUI screenshot of the aggregate interface configuration, which would be helpful for us to reproduce your issue?

Thanks,
Maxx

**Comment by dimk21** [comment]
> Hi @dimk21 ,
> 
> Thank you for bringing this issue to our attention. It appears that FOS is unable to find the interface named "LAN." I noticed that you are using an alias, which is intended for human readability. When FOS refers to the interface, we need to use the actual name instead of the alias. Could you kindly check if "LAN" is indeed the name of the interface you wish to use? On my side, I was able to create a VLAN interface under a VDOM by using a different VDOM aggregate interface in global mode. If the issue persists, could you please provide a GUI screenshot of the aggregate interface configuration, which would be helpful for us to reproduce your issue?
> 
> Thanks, Maxx

Hi @MaxxLiu22 ,
Thanks.
"LAN" is indeed the name of the interface I wish to use. See:
![image](https://github.com/user-attachments/assets/8e70e9fe-89a6-45d4-96e2-8717b6139055)

May I provide more Information for you to reproduce the issue? I know there is only one interface inside the aggregate interface. This will change in the future.

Kind regards

**Comment by alagoutte** [comment]
the interface is on root VDOM and not global VDOM.

**Comment by MaxxLiu22** [comment]
Hi @dimk21 ,

Thank you for providing the information. Your configuration looks good to me. I have created the same aggregate interface and VLAN via Ansible without encountering any errors.
![image](https://github.com/user-attachments/assets/4a4eb538-9599-4725-98c7-0b181c4831ba)


We have recently released Ansible FOS 2.3.7. Could you kindly upgrade it via ``ansible-galaxy collection install fortinet.fortios:2.3.7`` and try again at your convenience? If the issue persists, we may need to debug from the FOS side. Please run the following code on your FGT, then execute your Ansible script again to observe what FOS does after receiving the Ansible script.
```
config global
diagnose debug reset
diagnose debug cli 8
diagnose debug enable
```
and many thanks to @alagoutte for your time and help.

Thanks,
Maxx

**Comment by dimk21** [comment]
> Hi @dimk21 ,
> 
> Thank you for providing the information. Your configuration looks good to me. I have created the same aggregate interface and VLAN via Ansible without encountering any errors. ![image](https://private-user-images.githubusercontent.com/98774290/349217396-4a4eb538-9599-4725-98c7-0b181c4831ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNTgzMTQsIm5iZiI6MTcyMTE1ODAxNCwicGF0aCI6Ii85ODc3NDI5MC8zNDkyMTczOTYtNGE0ZWI1MzgtOTU5OS00NzI1LTk4YzctMGIxODFjNDgzMWJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDE5MjY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MTI0ZDY5NDI3ZDAyNTU1NWRjYjcxZjEzYTVkMzRkNWZlMThjZDllMDc0MjQ2ZWY5NjFhZDgzMWE2NTE2ZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FsocpcVltBfOFj1MQPnQ2sjR6omIHm4rnO8576RgWcw)
> 
> We have recently released Ansible FOS 2.3.7. Could you kindly upgrade it via `ansible-galaxy collection install fortinet.fortios:2.3.7` and try again at your convenience? If the issue persists, we may need to debug from the FOS side. Please run the following code on your FGT, then execute your Ansible script again to observe what FOS does after receiving the Ansible script.
> 
> ```
> config global
> diagnose debug reset
> diagnose debug cli 8
> diagnose debug enable
> ```
> 
> and many thanks to @alagoutte for your time and help.
> 
> Thanks, Maxx

Hi @MaxxLiu22 ,

thanks again.
This is what FOS does after receiving the Ansible script:

**0: config vdom
0: edit vdom-cust
0: config system interface
0: edit "INT_3930"
0: set vdom "vdom-cust"
0: set vrf 24
0: set ip 10.0.0.1 255.255.255.254
0: set allowaccess ping
0: set bfd enable
0: set bfd-desired-min-tx 750
0: set bfd-required-min-rx 750
0: set status down
0: set description "test_cust"
0: set alias "test_cust"
0: set role lan
0: set interface "LAN"
0: set vlanid 3930
cmd=config vdom
edit vdom-cust
config system interface
edit INT_3930
set vdom vdom-cust
set vrf 24
set ip 10.0.0.1 255.255.255.254
set allowaccess ping
set bfd enable
set bfd-desired-min-tx 750
set bfd-required-min-rx 750
set status down
set description test_cust
set alias test_cust
set role lan
set interface LAN
set vlanid 3930
abort

-3: end
[cmf_shm_update:663] Error: pid=3405,vd=vsys_hamgmt,query=system.settings,level=0,pos=0.**

We are currently running FOS 2.3.6
Maybe 2.3.7 will fix this.

Kind regards

**Comment by dimk21** [comment]
Hey @MaxxLiu22 ,

just updated to 2.3.7 - still the same issue. Seems like im doing something wrong.

Kind regards

**Comment by dimk21** [comment]
> Hi @dimk21 ,
> 
> Thank you for providing the information. Your configuration looks good to me. I have created the same aggregate interface and VLAN via Ansible without encountering any errors. ![image](https://private-user-images.githubusercontent.com/98774290/349217396-4a4eb538-9599-4725-98c7-0b181c4831ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxOTU3MTMsIm5iZiI6MTcyMzE5NTQxMywicGF0aCI6Ii85ODc3NDI5MC8zNDkyMTczOTYtNGE0ZWI1MzgtOTU5OS00NzI1LTk4YzctMGIxODFjNDgzMWJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDA5MjMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlMjEzMjBiMTljYTdmNjIxMmMzZGE0ZjdiYzBhODkxZDVkZjIwNjk2MTAwNDk1OTY1MzQ3MDY1MzI3Nzg3Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d5m_yqGGu__k1An_QkIoaaN0d7Nb70u77XKw8TmQY5Y)
> 
> We have recently released Ansible FOS 2.3.7. Could you kindly upgrade it via `ansible-galaxy collection install fortinet.fortios:2.3.7` and try again at your convenience? If the issue persists, we may need to debug from the FOS side. Please run the following code on your FGT, then execute your Ansible script again to observe what FOS does after receiving the Ansible script.
> 
> ```
> config global
> diagnose debug reset
> diagnose debug cli 8
> diagnose debug enable
> ```
> 
> and many thanks to @alagoutte for your time and help.
> 
> Thanks, Maxx

Hi @MaxxLiu22 ,

could you provide your Ansible script, so I could compare with mine. I still have no clue what is going wrong.
THX

**Comment by dimk21** [comment]
Finally fixed it!
My "api-profil" under Admin Profiles was "set scope vdom". Changed it to "set scope global". Now its working. Ty all for helping
